powerauth-push-client/src/main/java/io/getlime/push/client/PushServerClient.java

```diff
@@ -758,10 +760,21 @@ public Response removeAndroid(Long id) throws PushServerClientException {
         if (response.getStatus() == 200) {
             return jacksonObjectMapper.readValue(response.getRawBody(), typeReference);
         } else {
-            com.fasterxml.jackson.databind.ObjectMapper mapper = new com.fasterxml.jackson.databind.ObjectMapper();
-            ErrorResponse errorResponse = mapper.readValue(response.getRawBody(), ErrorResponse.class);
-            throw new PushServerClientException(response.getStatusText(), errorResponse.getResponseObject());
+            String responseBody = null;
+            try (Scanner scanner = new Scanner(response.getRawBody(), StandardCharsets.UTF_8.name()).useDelimiter("\\A")) {
+                if (scanner.hasNext()) {
+                    responseBody = scanner.next();
+                }
+            }
+            if (responseBody != null && !responseBody.isEmpty()) {
+                // Response body contains data, return Exception with status code and error response
+                com.fasterxml.jackson.databind.ObjectMapper mapper = new com.fasterxml.jackson.databind.ObjectMapper();
+                ErrorResponse errorResponse = mapper.readValue(responseBody, ErrorResponse.class);
```

Comment:
... this can probably fail in case there is some error HTML code, we should check for valid JSON. This will also handle the empty response (not valid JSON), so the error handling can be unified.